### ISSUE TITLE
Illegal offset in array check

### ISSUE THREAD

**Issue by danieltj27** [danieltj27/Custom-Archives#4]
```
[22-Jun-2018 13:51:27 UTC] PHP Warning:  Illegal offset type in isset or empty in /wp-content/plugins/custom-archives/custom-archives.php on line 668
[22-Jun-2018 13:51:27 UTC] PHP Warning:  Illegal offset type in isset or empty in /wp-content/plugins/custom-archives/custom-archives.php on line 620
```

I'm getting this error in the error.log file whenever I load the site, so it seems.

Looks like the `type` check is failing under certain conditions which should be addressed.

**Comment by danieltj27** [comment]
For reference; [line 620](https://github.com/danieltj27/Custom-Archives/blob/master/custom-archives.php#L620) and [line 668](https://github.com/danieltj27/Custom-Archives/blob/master/custom-archives.php#L668).

The error is in this block here:
```
			// Get the archive post type
			$post_type = $wp_query->query['post_type'];
			// Get the page id for the archive page
			$post_id = ( isset( $ids[ $post_type ] ) ) ? $ids[ $post_type ] : 0;
```

If `$wp_query->query['post_type']` isn't set, `$post_type` become null and checking it then breaks the `isset` check later on. Either the `$post_type` line needs a fallback value that will always gracefully fail or the `$post_id` line needs protection from a null value.

Seems better to fix the type check and provide a false fallback in this scenario.

**Comment by danieltj27** [comment]
Can't seem to reproduce the error. Will have to investigate and come back to it.

**Comment by danieltj27** [comment]
This should be fixed in commit 9cca31d. Will reopen if I spot the error again.
